src/mmw/js/src/modeling/models.js

```diff
@@ -1345,6 +1350,13 @@ var ScenarioModel = Backbone.Model.extend({
         return taskModel.start(taskHelper);
     },
 
+    queueFetchResultsIfNeeded: function() {
+        if (this.fetchResultsPromise) {
+            return this.fetchResultsPromise.always(this.fetchResultsIfNeeded);
```

Comment:
It seems like the intent of this line is to queue up `fetchResults` requests, but currently, since `this.fetchResultsPromise` is not updated, if `queueFetchResultsIfNeeded` is called multiple times before the first promise resolves, all subsequent fetches will be queued up after the first one, and be fired simultaneously rather than in order.

To fire them in order, prefer doing something like this:

```js
if (this.fetchResultsPromise) {
  this.fetchResultsPromise = this.fetchResultsPromise.then(this.fetchResultsIfNeeded, this.fetchResultsIfNeeded);
  return this.fetchResultsPromise;
}
```

Also note that `.always()` returns the parent promise, while `.then()` returns a new one, which is what we want since the next time we'll queue it after the latest one, not the first one. https://stackoverflow.com/a/32882576/6995854